src/test/java/com/google/sps/CommentAnalysisTest.java

```diff
@@ -0,0 +1,79 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.sps.servlets.CommentAnalysis;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public class CommentAnalysisTest {
+  private CommentThread testCommentThread;
+  private static final String APPLICATION_NAME = "testComment";
+  private static final String DEVELOPER_KEY = "";
+  private static final String TEST_VIDEO_ID = "E_wKLOq-30M";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /**
+   * Build and return an authorized API client service.
+   *
+   * @return an authorized API client service
+   * @throws GeneralSecurityException, IOException
+   */
+  public static YouTube getService() throws GeneralSecurityException, IOException {
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, JSON_FACTORY, null)
+               .setApplicationName(APPLICATION_NAME)
+               .build();
+  }
+
+  @Before
+  public void setUp() throws GeneralSecurityException, IOException{
+      YouTube youtubeService = getService();
+      YouTube.CommentThreads.List youtuberequest = youtubeService.commentThreads().list("snippet");
+      CommentThreadListResponse youtuberesponse = youtuberequest.setKey(DEVELOPER_KEY)
+                                                      .setVideoId(TEST_VIDEO_ID)
+                                                      .setMaxResults(2L)
+                                                      .setTextFormat("plainText")
+                                                      .execute();
+      testCommentThread = youtuberesponse.getItems().get(0);
+  }
+
+  @Test
+  public void testSentimentAnalysisInRange () {
+    // Test the Sentiment Analysis Score within range -1 to 1.
+    CommentAnalysis analysis = new CommentAnalysis(testCommentThread);
+    try {
+      float result = analysis.sentiAnalysis();
+      Assert.assertTrue(result >= -1 && result <= 1);
+    } catch (IOException e) {
+      Assert.assertTrue("Catch Exception:" + e.getMessage(), false);
+    }
```

Comment:
Should we be catching any thrown exceptions here?

It seems that if the analysis.sentiAnalysis() method invocation throws an exception we should likely just propogate it. This would likely match our expectations that if the sentiment analsysis throws an exception in our unit test, the test should fail.